docs/template-variables.rst

```diff
@@ -0,0 +1,393 @@
+
+__ https://getnikola.com/theming.html#identifying-and-customizing-different-kinds-of-pages-with-a-shared-template
+
+Variables available in post pages (``post.tmpl``, ``story.tmpl`` etc.)
+----------------------------------------------------------------------
+
+.. class:: table table-bordered table-striped
+
+======================  ==========  ========================================================
+Name                    Type        Description
+======================  ==========  ========================================================
+``post``                Post        The post object
+``permalink``           str         Permanent link to the post
+``enable_comments``     bool        True for posts, ``COMMENTS_IN_PAGES`` setting for pages
+======================  ==========  ========================================================
+
+Variables available in post lists
+---------------------------------
+
+.. class:: table table-bordered table-striped
+
+==============  =============  ==============================================
+Name            Type           Description
+==============  =============  ==============================================
+``posts``       list<Post>     List of post objects that appear in this list
+``prevlink``    str            Link to previous page
+``nextlink``    str            Link to next page
+==============  =============  ==============================================
+
+
+Variables available in indexes
+------------------------------
+
+.. class:: table table-bordered table-striped
+
+==============================  ==============  ===============================================================================
+Name                            Type            Description
+==============================  ==============  ===============================================================================
+``posts``                       list<Post>      List of post objects that appear in this list
+``index_teasers``               bool            ``INDEX_TEASERS`` setting
+``show_index_page_navigation``  bool            ``SHOW_INDEX_PAGE_NAVIGATION`` setting
+``current_page``                int             Number of current page
+``page_links``                  list<str>       Links to different pages
+``prevlink``                    str             Link to previous page
+``nextlink``                    str             Link to next page
+``prevfeedlink``                str             Link to previous page as an Atom feed
+``nextfeedlink``                str             Link to next page as an Atom feed
+``prev_next_links_reversed``    bool            Whether or not previous and next links should be reversed (``INDEXES_STATIC``)
+==============================  ==============  ===============================================================================
+
+Variables available in taxonomies
+---------------------------------
+
```

Comment:
More generally, it would be good to have a list of which templates are used by which parts of Nikola. This is also helpful for people who want to create a template from scratch instead of deriving from the base templates.